packages/eslint-plugin/src/rules/member-ordering.ts

```diff
@@ -615,6 +615,39 @@ function getRank(
   return getRankOrder(memberGroups, orderConfig);
 }
 
+function changeToGroups(
+  memberSet: Member[],
+  memberType: MemberType[],
+  supportsModifiers: boolean,
+): Member[][] {
+  const groupedMembers: Member[][] = [];
+
+  const getPreviousRank = (): number | undefined => {
```

Comment:
[Refactor] This function is constantly calling `getRank` and `.at(...)` to recompute things. But I think they can be cached in variables:

* Instead of calling `getRank(member, memberType, supportsModifiers)` twice, store in a variable
* Instead of a dynamic `getPreviousRank`, have a nullable state variable like `let previousGroupAndRank: { group: Member[]; rank: number } | undefined;`

Does that make sense?

Reply:
Thank you so much for the reviews.

But, can you explain to me the 2nd point a little bit more ? I can't seem to understand that.

Reply:
Ah sorry for not being clear!

What I'm looking at: right now, `getPreviousRank()` is being called in each iteration of `memberSet.forEach(...)`. It's essentially recomputing a value each time it's called, right? Where that "value" is the previous rank.

Problem (I think, unless I've misinterpreted): Wasn't that rank computed for the previous member? If, say, we're at `forEach` index 3, then `getPreviousRank()` will try to return the rank for the member at index 2. But didn't we compute the rank for member at index 2 in the previous iteration of the `forEach` loop?

My proposal: instead of recomputing the previous rank, storing it in a variable. That way the code is a little more clear and the act of computing a member's rank isn't done so many times per member.

Reply:
Ah sorry for not being clear!

What I'm looking at: right now, `getPreviousRank()` is being called in each iteration of `memberSet.forEach(...)`. It's essentially recomputing a value each time it's called, right? Where that "value" is the previous rank.

Problem (I think, unless I've misinterpreted): Wasn't that rank computed for the previous member? If, say, we're at `forEach` index 3, then `getPreviousRank()` will try to return the rank for the member at index 2. But didn't we compute the rank for member at index 2 in the previous iteration of the `forEach` loop?

My proposal: instead of recomputing the previous rank, storing it in a variable. That way the code is a little more clear and the act of computing a member's rank isn't done so many times per member.

Does that make sense?

Reply:
> My proposal: instead of recomputing the previous rank, storing it in a variable. That way the code is a little more clear and the act of computing a member's rank isn't done so many times per member.

Yes, thanks :raised_hands: